src/sigma/sigmaplus_proof.h

```diff
@@ -51,6 +51,20 @@ class SigmaPlusProof {
     R1Proof<Exponent, GroupElement> r1Proof_;
     std::vector<GroupElement> Gk_;
     Exponent z_;
+
+public:
+    bool operator==(const SigmaPlusProof& other) const {
+        return n == other.n &&
+            m == other.m &&
+            B_ == other.B_ &&
+            r1Proof_ == other.r1Proof_ &&
+            Gk_ == other.Gk_ &&
+            z_ == other.z_;
+    }
+
+    bool operator!=(const SigmaPlusProof& other) const {
+        return !(*this == other);
+    }
```

Comment:
IMO this one should come after the constructors and before the class-specific methods.